1_fetch/src/get_sb_data.R

```diff
@@ -131,10 +131,17 @@ prep_feature_vars <- function(sb_var_data, sites, retain_vars) {
   #'@return data frame with COMID column appended by all feature variables of interest; 
   #'time-varying features converted to long-term averages where applicable
   
+  options(dplyr.summarise.inform = FALSE)
+  
   data <- sites %>%
     select(COMID, all_of(retain_vars)) %>%
     mutate(across(where(is.character), as.numeric))
   
+  if("ID" %in% retain_vars) {
+    data <- rename(data, GAGES_ID = ID) %>%
+      mutate(GAGES_ID = as.character(GAGES_ID))
+  }
+  
   #join all sciencebase data to single data frame with comids
   for (i in 1:length(sb_var_data)) {
     filepath <- sb_var_data[[i]][1]
```

Comment:
This will also have to be edited because `summarise_all` will try to summarize the GAGES_ID. I'm not sure of a general way to handle this problem, but a quick fix is to only do this step `if(!("ID" %in% retain_vars))`
```
 #handles any duplicated COMIDs
  data <- data %>% 
    group_by(COMID) %>%
    summarise_all(mean)
```